far/viewer.cpp

```diff
@@ -130,12 +130,15 @@ Viewer::Viewer(window_ptr Owner, bool bQuickView, uintptr_t aCodePage):
 	ViOpt(Global->Opt->ViOpt),
 	Reader(ViewFile, (Global->Opt->ViOpt.MaxLineSize*2*64 > 64*1024 ? Global->Opt->ViOpt.MaxLineSize*2*64 : 64*1024)),
 	strLastSearchStr(Global->GetSearchString()),
-	LastSearchCaseSensitive(Global->GlobalSearchCaseSensitive),
-	LastSearchWholeWords(Global->GlobalSearchWholeWords),
-	LastSearchReverse(Global->GlobalSearchReverse),
-	LastSearchHex(Global->GetSearchHex()),
-	LastSearchRegexp(Global->Opt->ViOpt.SearchRegexp),
-	LastSearchDirection(Global->GlobalSearchReverse? -1 : +1),
```

Comment:
@MKadaner looks like this one was here for a reason: reverse search doesn't work as expected now.

STR:
- view a file with multiple occurrences of some pattern
- F7, enter it, enter
- Shift+F7 a few times
- Alt+F7 once. It goes back.
- Alt+F7 again. It goes forward.
- Alt+F7 again a few times. It loops.
- F7. Note that `[x] Reverse search` is selected.

Restoring this variable fixes the issue.

Reply:
No problem, let's wait till the weekend.

Reply:
Done. See #599.